cluster-up.sh

```diff
@@ -0,0 +1,223 @@
+#! /bin/bash
+#
+# 'gk-up.sh', and its companion startup script 'do-startup.sh', create a gce cluster customized using a
+# rhel 7 image with gluster, heketi, git, helm, kubernetes, service-catalog, etc installed. The startup
+# script creates a file named "/root/__SUCCESS" once it has completed. If the gce cluster is running it
+# will be shut down (this script is idempotent.) A custom cns s3 service broker is also installed.
+#
+# Args:
+#		-y to bypass hitting a key to acknowlege seeing the displayed config info.
+#       The default is to prompt the user to continue.
+#		-i Deploy GCE instances and stop.  Startup script will still install packages and dependencies on all VMs.
+#          DO NOT deploy k8s, gluster-kubernetes (including S3 component), or the CNS Objet Broker
+#		-g Deploy gluster-kubernetes and stop.
+#          DO NOT install S3 components or the CNS Object Broker
+#		-s Deploy gluster-kubernetes, S3 components and stop.
+#          DO NOT install CNS Object Broker
+# Note: lowercase vars are local to functions. Uppercase vars are global.
+
+set -euo pipefail
+
+# Print help
+function print_help() {
+	echo "Gluster-Kubernetes GCE deployment script."
+	echo "-y to bypass hitting a key to acknowlege seeing the displayed config info."
+	echo "   The default is to prompt the user to continue."
+	echo "-i Deploy GCE instances and stop.  Startup script will still install packages and dependencies on all VMs."
+	echo "   DO NOT deploy k8s, gluster-kubernetes (including S3 component), or the CNS Objet Broker"
+	echo "-g Deploy gluster-kubernetes and stop."
+	echo "   DO NOT install S3 components or the CNS Object Broker"
+	echo "-s Deploy gluster-kubernetes, S3 components and stop."
+	echo "   DO NOT install CNS Object Broker"
+}
+
+# TODO add provider flag
+# Parse out -y arg if present.
+function parse_args() {
+	# 0=true, 1=false
+	DO_CONFIG_REVIEW=0
+	INSTALL_GLUSTER=0
+	INSTALL_OBJECT=0
+	INIT_KUBE=0
+	while getopts yYnNigsh o; do
```

Comment:
may want to add --long equivalent args later...

Reply:
It'd be nice to have args for most of the config options.  Stackoverflow seems to dislike `getopt` for lack of portability / non-uniform behavior across shells.  What's your take on it?  I'm sure for us it wouldn't be a problem given we all use a RHEL variant to work in, so we've all got a recent version of bash.

Reply:
I looked up `getop` too and found similar comments about it's lack of portability, etc. But getopt*s* does not support long option names which I think is a problem. There are several comments and notes about the _newer_ version of `getopt` (singular) being much improved and a now a viable solution. For now I like `getopt` better.